lib/Doctrine/DBAL/Platforms/MySqlPlatform.php

```diff
@@ -650,6 +653,31 @@ public function getAlterTableSQL(TableDiff $diff)
         return array_merge($sql, $tableSql, $columnSql);
     }
 
+    protected function getColumnDeclarationPositionSQL(string $columnName, ?Table $toTable): ?string
+    {
+        if ($toTable === null) {
+            return null;
+        }
+
+        $columnFound = false;
+        $prevColumn  = null;
+        foreach ($toTable->getColumns() as $k => $column) {
+            if ($k === $columnName) {
+                $columnFound = true;
+
+                break;
+            }
+
+            $prevColumn = $column;
+        }
+
+        if (! $columnFound) {
+            throw new InvalidArgumentException('Column name "' . $columnName . '" not found');
```

Comment:
Would it be possible to avoid this exception happening by using a more suitable method signature? Each column is either first or goes after some other column. See https://github.com/doctrine/dbal/pull/4145 for example. Otherwise, this will need a functional test that demonstrates the exception being thrown.

Reply:
we can call `$toTable->getColumn()` before the foreach loop which should check if the column exists or not

please sudgest ideal approach to this

Reply:
It might work if instead of having the `$toTable` in the `SchemaDiff`, the `ColumnDiff` had a property like `$afterColumn` that would contain the column name and be non-empty only if the column position changed. This assumes that a similar SQL syntax could be used on other supported platforms. This would also solve the problem of having the `FIRST` and `AFTER` fragments unnecessarily generated.

Reply:
`ColumnDiff` is used also for a new columns?

Reply:
`ColumnDiff` can not be used as we need position also for `Column`. @morozov I would like to finish this PR asap, please advise, if the `$toTable` is ok - used like `$fromTable` already used - or if you have a better alternative. Thanks.

Reply:
> ColumnDiff can not be used as we need position also for Column

I do not understand what this means. Can you please elaborate?

Reply:
> 
> 
> > ColumnDiff can not be used as we need position also for Column
> 
> I do not understand what this means. Can you please elaborate?

if column already exists, then `ColumnDiff` instance is returned from the comparator

if the column is new (not renamed/modified), then there is no `ColumnDiff` for it

Reply:
exception is covered by added test

Reply:
@mvorisek please make sure you don't resolve the conversations that you didn't start unless you're resolving one on the starter's behalf.